microbench-ci: combine compare and post

herkolategan opened the pull request:
Previously, the microbench-ci tool had separate commands for comparing benchmark results and posting them to GitHub.

This change combines the compare and post functionality into a single command, simplifying the CI workflow by reducing the number of commands needed. When `--post` is specified, the tool will both generate the comparison summary and conditionally post it as a GitHub comment.

Previously, a single comment was posted or updated on the PR for each invocation of the tool. Now, a new comment will be posted for each invocation of the tool if either the `X-perf-check` label is present or if the benchmark results have significant changes.

Epic: None
Release note: None

herkolategan committed:
microbench-ci: combine compare and post

Previously, the microbench-ci tool had separate commands for comparing benchmark
results and posting them to GitHub.

This change combines the compare and post functionality into a single command,
simplifying the CI workflow by reducing the number of commands needed. When
`--post` is specified, the tool will both generate the comparison summary and
conditionally post it as a GitHub comment.

Previously, a single comment was posted or updated on the PR for each invocation
of the tool. Now, a new comment will be posted for each invocation of the tool
if either the `X-perf-check` label is present or if the benchmark results have
significant changes.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/142469)
<!-- Reviewable:end -->

herkolategan committed:
microbench-ci: remove description from github summary

The description on the summary is no longer needed, because we now add a label
if a significant change was detected.

Additionally, the legend is not needed, because there is no longer a scenario
where we need to indicate only a possibly regression.

herkolategan committed:
github, microbench-ci: skip perf check if label is present

If the 'X-skip-perf-check' label is present, skip the performance check.

Epic: None
Release note: None

herkolategan committed:
microbench-ci: relax thresholds

Relax the thresholds a bit based on data collected from CI runs.

Epic: None
Release note: None
